examples/resources/gitlab_runner/resource.tf

```diff
@@ -0,0 +1,44 @@
+# Basic GitLab Runner
+resource "gitlab_runner" "this" {
+  token = "12345"
+}
+
+# GitLab Runner that runs only tagged jobs
+resource "gitlab_runner" "tagged_only" {
+  token       = "12345"
+  description = "I only run tagged jobs"
+
+  run_untagged = "false"
+  tag_list     = ["tag_one", "tag_two"]
+}
+
+# GitLab Runner that only runs on protected branches
+resource "gitlab_runner" "protected" {
+  token       = "12345"
+  description = "I only run protected jobs"
+
+  access_level = "ref_protected"
+}
+
+# Generate a `config.toml` file that you can use to create a runner
+# This is the typical workflow for this resource, using it to create an authentication_token which can then be used
+# to generate the `config.toml` file to prevent re-registering the runner every time new hardware is created.
+resource "gitlab_runner" "my_runner" {
+  token = "12345"
+}
+
+# This creates a local "shell" runner, but can be changed to generate whatever is needed.
```

Comment:
Correct if I'm wrong but this won't really create a functional runner, unless you place that file in a cloud instance (for example).

I have a actual runner using the [official GitLab runner Helm chart](https://gitlab.com/gitlab-org/charts/gitlab-runner). Would you like me to add as an example?

Reply:
No, I'd just registers one to obtain the auth token - nothing more :) 

I guess feel free to provider the example and see how it can fit the docs ...

Reply:
I wasn't actually trying to create a fully functioning runner in the example, honestly, because the terraform to do so in a secure fashion is... large. I was trying to provide juuuuust enough that someone could see "oh, this is how this resource is meant to be used".

Using your helm chart example, to create a functioning runner we'd have to pull in tf for creating a kubernetes cluster too, right? It feels like that's out of scope of the documentation personally.

Reply:
> It feels like that's out of scope of the documentation personally

I agree it's out of scope, but in another hand I think we need to provide a more realistic example.
Would it look stupid to reference the chart but skip the Kubernetes cluster creation? Looks like a middle term.

Reply:
@willianpaixao  - would you be willing to provide some TF that deploys that so I can include it in the PR? I think that makes sense, but I've not installed the runner via a helm chart (yet 😄 ), so I'd hate to include a bad example. It looks like the helm chart terraform provider accepts a kube config, so we can certainly leave setting up a k8s cluster as an 'exercise for the reader'.

Reply:
@willianpaixao  - I did some digging on this last night, and I don't think the helm chart would actually work with this resource, since there doesn't seem to be a way to short-circuit specifying the registration token and directly supply the authentication token. 

Reference on registration token being required: https://docs.gitlab.com/runner/install/kubernetes.html#required-configuration

You could potentially specify the token directly as it's injected into the runner here: https://docs.gitlab.com/runner/install/kubernetes.html#additional-configuration

I'd suggest though that until we can have someone test that, we should stick with the current documentation and add that additional piece later as needed.

Reply:
> I'd suggest though that until we can have someone test that, we should stick with the current documentation and add that additional piece later as needed.

Yes, I agree. What we already have is plenty w.r.t to docs. Let's move forward and add more documentation in another iteration 🐾